### ISSUE TITLE
transformTagName and transformAttributeName aren't typed for Builder.

### ISSUE THREAD

**Issue by raarts** [NaturalIntelligence/fast-xml-parser#524]
* [x] Are you running the latest version?
* [x] Have you included sample input, output, error, and expected output? 
* [x] Have you checked if you are using correct configuration?
* [x] Did you try online [tool](https://naturalintelligence.github.io/fast-xml-parser/)?

### Description
TypeScript gives an error when trying to add  transformTagName or transformAttributeName to the options

Side issue: there a small bug in[ OrderedObjsParser](https://github.com/NaturalIntelligence/fast-xml-parser/blame/master/src/xmlparser/OrderedObjParser.js#L267), there's a blank between `this.` and `options`. JS is forgiving so doesn't break anything, but still.
### Input
n/a

### Code
```js
  const builder = new XMLBuilder({
    ignoreAttributes: false,
    attributeNamePrefix: '',
    attributesGroupName: 'attrs',
    transformTagName: (tag) => {
    // ^^^ Object literal may only specify known properties, and 'transformTagName' does not exist in type 'Partial<XmlBuilderOptions>'.
      const newName = json2xml.get(tag);
      if (newName) return newName;
      return tag;
    },
    transformAttributeName: false,
  });
```

### expected data

**Would you like to work on this issue?**

- [x] Yes
- [ ] No

**Comment by raarts** [comment]
I found another bug: textnodes are suddenly being created when I use transformTagName. Tracing what happens, a question arises: why aren't transformTagName (and removeNSPrefix) implemented in compress(), *just* before the JSON property is created? That would be the logical location.

**Comment by amitguptagwl** [comment]
@raarts `transformAttributeName` and `transformTagName` are not the options for builder but for parser only. `compress` is an internal method the purpose of that to prettify the ordered json into normal json. Because ordered json is big list of nodes. 

Can you please add more detail on "I found another bug: textnodes are suddenly being created when I use transformTagName. "?

**Comment by raarts** [comment]
Indeed. I found it in the source code in the Builder options, but it doesn't do anything.

About the other bug: I was using parseTagValue, but when I added transformTagName suddenly #text nodes again were generated. I can't reproduce it any more, since I switched to xmls-js, I needed to import and then re-export while translating tags and attribute names. Sorry for wasting your time, I should have closed this bug report.
